src/code_109660_len_1270.c

```diff
@@ -1,37 +1,231 @@
 #include "common.h"
+#include "map.h"
+
+MapConfig* get_current_map_header(void);
+void disable_player_input(void);
+void disable_player_static_collisions(void);
+void move_player(s32, f32, f32);
+s32 func_802E854C();
+
+typedef struct struct802E7DE0 {
+    /* 0x000 */ s32 unk_000;
+    /* 0x004 */ s32 unk_004;
+    /* 0x008 */ u8* unk_008;
+    /* 0x00C */ s32 unk_00C;
+    /* 0x010 */ u8 unk_010[36];
+    /* 0x034 */ s8 unk_034[36];
+    /* 0x058 */ u8 unk_058[36];
+    /* 0x07C */ u8 unk_07C[36];
+    /* 0x0A0 */ u8 unk_0A0[36];
+    /* 0x0C4 */ f32 unk_0C4[36];
+    /* 0x154 */ f32 unk_154[36];
+    /* 0x1E4 */ f32 unk_1E4[36];
+    /* 0x274 */ f32 unk_274[36];
+    /* 0x304 */ f32 unk_304[36];
+} struct802E7DE0;
 
 INCLUDE_ASM(s32, "code_109660_len_1270", func_802E7DE0);
 
-INCLUDE_ASM(s32, "code_109660_len_1270", func_802E7F40);
+typedef struct struct802E7F40 {
+    /* 0x00 */ s32 unk_00;
+    /* 0x04 */ u16 unk_04;
+    /* 0x06 */ char unk_06[6];
+    /* 0x0C */ f32 unk_0C;
+    /* 0x10 */ char unk_10[4];
+    /* 0x14 */ s32 unk_14;
+} struct802E7F40;
 
-INCLUDE_ASM(s32, "code_109660_len_1270", func_802E7F6C);
+void func_802E7F40(Entity* entity) {
+    struct802E7F40* temp = entity->dataBuf;
+
+    entity->scale.y = 0.85714287f;
+    temp->unk_04 = 0xFFFF;
+    temp->unk_00 = D_8015C7D0;
+}
+
+void func_802E7F6C(Entity* entity) {
+    struct802E7F40* temp = entity->dataBuf;
+
+    temp->unk_0C = entity->position.y;
+    func_802E7DE0(entity, &D_0A004350, &D_0A0026F0);
+}
 
 INCLUDE_ASM(s32, "code_109660_len_1270", func_802E7FA0);
 
+extern f64 D_802EB370;
+
 INCLUDE_ASM(s32, "code_109660_len_1270", func_802E854C);
 
-INCLUDE_ASM(s32, "code_109660_len_1270", func_802E8858);
+s32 func_802E8858(Entity* entity) {
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    s32 phi_a0 = 0;
 
-INCLUDE_ASM(s32, "code_109660_len_1270", func_802E88EC);
+    if (entity->unk_06 & 1) {
+        if ((playerStatus->actionState == ActionState_GROUND_POUND) || (playerStatus->actionState == ActionState_ULTRA_POUND)) {
```

Comment:
Are these extra parens needed?

Reply:
it can change how this is evaluated in some cases. It's generally good practice.